core/services/relay/evm/mercury/v1/data_source.go

```diff
@@ -291,39 +290,34 @@ func (ds *datasource) executeRun(ctx context.Context) (*pipeline.Run, pipeline.T
 }
 
 func (ds *datasource) setLatestBlocks(ctx context.Context, obs *relaymercuryv1.Observation) {
-	latestBlocks := ds.getLatestBlocks(ctx, nBlocksObservation)
+	latestBlocks, err := ds.chainReader.LatestHeads(ctx, nBlocksObservation)
+	if err != nil {
+		ds.lggr.Errorw("failed to read latest blocks", "error", err)
```

Comment:
Might choose to panic here instead since we never expect this to happen

Reply:
Sometimes a plugin will crash or be unavailable. This is expected to happen during "normal" operation and must be handled gracefully.

Reply:
We can bubble it up if we think this should result in a pipeline execution error, but this is a change in the current behaviour of error handling and might deserve more attention. If so, we could revert the related changes in the chainlink-relay to unblock the build of core images.

Reply:
In that case, can we make this method return error and error out in the calling function also?

Reply:
by that you mean to Observe() to error out on  setLatestBlocks() error ?